### ISSUE TITLE
Tests compilation

### ISSUE THREAD

**Issue by Sanjo** [meteor/meteor#5427]
TLDR: Compiling the tests into a separate build target would enable me (Velocity) to speed up the test execution a lot.

With our current mirror approach in Velocity, we run the Meteor tool separately for each mirror (= testing mode). On average you use three mirrors. One for Cucumber, Jasmine server tests and Jasmine client tests. That means that four separate Meteor tool instances run for the app.
With the current approach therefore a lot of time is spend on building the same app (four times).

The only difference between the mirrors is the set of tests that are included. Therefore we could build the app only once and then spawn the built app directly and load some tests based on an argument that we pass to the programs/server/main.js.

My sticking point right now is that the tests should get compiled, so that our users can us ES2015 and CoffeeScript. I tried to implement it, but have some hard time to get it fully working. So I would appreciate some help.

**Comment by tmeasday** [comment]
Hi @Sanjo, thanks for this one, sorry it's taken a while to look at it.

Is this still something that you want for your non-velocity projects? I've been considering all the various ways Velocity integrates with the tool and ideas on how to make that better and I'm working on (what I hope is a) really simple proposal to get a better app testing story going forward. I'll post something to the forums soon I hope.

But let me know your latest thoughts on this PR.

**Comment by Sanjo** [comment]
Yes. I still think the PR is a good idea. I enables me to write a fast implementation for server unit testing for sanjo:jasmine. The only bigger missing part right now for me.

**Comment by tmeasday** [comment]
So sanjo:jasmine somehow runs a mirror using the pre-built test code without the use of velocity?

**Comment by Sanjo** [comment]
Yeah, I would just spawn a modified version of `.meteor/local/build/programs/server/boot.js`. This doesn't have the overhead of the second Meteor build tool running.

**Comment by abernix** [comment]
Closing as I think this is no longer relevant or needs to be re-visited with the newer testing setups.
